### ISSUE TITLE
Possible Bug: Kodi crash on unsuccessful resolve, Feature Requests: Colors for Quality, Multi Subtitles, more...

### ISSUE THREAD

**Issue by cabban** [Tikipeter/repository.tikipeter#46]
hello again,and thank u for doing hard work on this lovely addon_=:)

io have 2 issues it has start chrashing when not open the link 
also it takes 10 seconds to arrive the window for scraper  :(
i hope some of this is fixable _=) cheers

**Comment by Tikipeter** [comment]
Hi cabban.

It's possible that the second issue (slow loading of results) was re-fixed with the most recent update (1.6.61a). I had reverted some changes that made things faster when I was trying to fix another issue (Results Display).

It's a little hard to understand what you are saying for the first issue, but I think you're saying that when a link fails to resolve, it crashes Kodi. Is that correct?

**Comment by cabban** [comment]
> Hi cabban.
> 
> It's possible that the second issue (slow loading of results) was re-fixed with the most recent update (1.6.61a). I had reverted some changes that made things faster when I was trying to fix another issue (Results Display).
> 
> It's a little hard to understand what you are saying for the first issue, but I think you're saying that when a link fails to resolve, it crashes Kodi. Is that correct?

Yes that is correct, when some scrapers fails to resolve, that makes kodi crash without a sound only take me to homedesk 😁 thank u 😀 I will test. But it took when I start using its tools 0.5-1 sec to find results

**Comment by Tikipeter** [comment]
For the Kodi crashing issue, you need to have it happen, and then go in and grab the log and put a link to it here. If you restart Kodi before getting the log, then you will need to get the kodi.old log file instead. I will hopefully be able to see if it's a Fen issue.

The other issues from your post are, unfortunately, going to have mainly negative answers. :( :(

* Openload/Streamango are no more. They have been shut down by authorities. I'll add them to the list of blocked sites on next update.

* Getting incorrect results are unfortunately the bi-product of the scrapers used. They will occasionally give wrong results. If there are any that consistently give false results, list them here and I will do my best to fix them. I am not good with editing scrapers though.

* Fen uses the fairly new opensubtitles API for requesting subs. The API is unfortunately limited to only one language. Perhaps I could make a second call with a second language chosen. Let me think about it some more.....

* Subscene and Podnapsi I will look into. If they don't provide an API I won't be able to add them. Also adding them would mean a re-write of the subtitle functionality, so it will not be a trivial addition. Let me think some more about this as well...

**Comment by cabban** [comment]
> For the Kodi crashing issue, you need to have it happen, and then go in and grab the log and put a link to it here. If you restart Kodi before getting the log, then you will need to get the kodi.old log file instead. I will hopefully be able to see if it's a Fen issue.
> 
> The other issues from your post are, unfortunately, going to have mainly negative answers. :( :(
> 
> * Openload/Streamango are no more. They have been shut down by authorities. I'll add them to the list of blocked sites on next update.
> * Getting incorrect results are unfortunately the bi-product of the scrapers used. They will occasionally give wrong results. If there are any that consistently give false results, list them here and I will do my best to fix them. I am not good with editing scrapers though.
> * Fen uses the fairly new opensubtitles API for requesting subs. The API is unfortunately limited to only one language. Perhaps I could make a second call with a second language chosen. Let me think about it some more.....
> * Subscene and Podnapsi I will look into. If they don't provide an API I won't be able to add them. Also adding them would mean a re-write of the subtitle functionality, so it will not be a trivial addition. Let me think some more about this as well...
> 
> EDIT: I forgot about the different colours for file quality. That is interesting. Leave it with me...

Thank u, then some ideas is greatt😀
ThanThank again
Yes some other addons has multilanguage. There mark. I wil show. And yes subtitles service is there is kodi settings subscene and podnapsi i have choosen 😀 see attached screenshouts 😀
![IMG_20191103_143035](https://user-images.githubusercontent.com/19514493/68085929-a1f71080-fe46-11e9-9dc4-fa057aa0154c.jpg)
![IMG_20191103_143014](https://user-images.githubusercontent.com/19514493/68085930-a1f71080-fe46-11e9-932c-32496a554caf.jpg)
![IMG_20191103_142937](https://user-images.githubusercontent.com/19514493/68085931-a28fa700-fe46-11e9-9861-8faf45918cb2.jpg)

**Comment by Tikipeter** [comment]
* Color highlights for results can now be applied by quality in latest update.

Longer wait on other feature requests.

**Comment by Tikipeter** [comment]
* Secondary Language can now be set for when Primary Language fails to give results.

Longer wait on other featrue requests.

**Comment by cabban** [comment]
> * Color highlights for results can now be applied by quality in latest update.
> 
> Longer wait on other feature requests.

thank u:)  where do i find the new settings , i have looked but cant find it :)

**Comment by Tikipeter** [comment]
Settings-->Results-->Display-->Highlight Results Based On...

Then you can go to 'Color Highlights' in the same tab and select the colors for the different qualities.

**Comment by Tikipeter** [comment]
* New setting in Playback->Subtitles to manually choose subs.

**Comment by cabban** [comment]
> * New setting in Playback->Subtitles to manually choose subs.

thank u best developer :)
